jbpm-test-coverage/src/test/java/org/jbpm/test/functional/CorrelationKeyTest.java

```diff
@@ -162,4 +168,53 @@ public void testMultiValuedKeyUniqueButInclusive() {
         ProcessInstance processInstance2 = ksession.startProcess(PROCESS, key2, null);
         assertProcessInstanceActive(processInstance2.getId());
     }
+
+    @Test
+    public void testCreateProcessInstanceDefaultValue() {
+        CorrelationKey key = keyFactory.newCorrelationKey(SIMPLE_KEY);
+
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        WorkflowProcessInstance processInstance = (WorkflowProcessInstance) ksession.startProcess(PROCESS, key,
+                parameters);
+
+        assertEquals("defaultProc", processInstance.getVariable("procVar"));
+        assertEquals(Integer.valueOf(1), processInstance.getVariable("secondVar"));
+
+    }
+
+    @Test
+    public void testCreateProcessInstanceWithDefaultValueAndParams() {
+        CorrelationKey key = keyFactory.newCorrelationKey(SIMPLE_KEY);
+
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        parameters.put(VARIABLE_ID, VARIABLE_VALUE);
+        WorkflowProcessInstance processInstance = (WorkflowProcessInstance) ksession.startProcess(PROCESS, key,
+                parameters);
+       
+        assertEquals(VARIABLE_VALUE, processInstance.getVariable("procVar"));
+        assertEquals(Integer.valueOf(1), processInstance.getVariable("secondVar"));
+
+    }
+
+    @Test
+    public void testCreateProcessInstanceComplexTypeDefaultValue() throws SAXException, IOException {
+
+        String address = "<org.jbpm.test.domain.Address><street>abc</street><number>29</number><city>def</city></org.jbpm.test.domain.Address>";
+
+        CorrelationKey key = keyFactory.newCorrelationKey(SIMPLE_KEY);
+
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        WorkflowProcessInstance processInstance = (WorkflowProcessInstance) ksession.startProcess(PROCESS, key,
+                parameters);
+
+        DetailedDiff diff = new DetailedDiff(
+                XMLUnit.compareXML(address, (String) processInstance.getVariable("complexVar")));
```

Comment:
It's not clear to me that this testing is working properly in the sense of returned variable is an instance of String. I was expecting to get a custom org.jbpm.test.domain.Address. For example, if the parameters map sets up a "complexVar" with an Address value, it's obtained again an object from this type, wdyt?